lib/ret_web/controllers/page_controller.ex

```diff
@@ -745,43 +781,105 @@ defmodule RetWeb.PageController do
       if is_cors_proxy_url do
         allowed_origins =
           Application.get_env(:ret, RetWeb.Endpoint)[:allowed_origins] |> String.split(",")
+
         opts =
           ReverseProxyPlug.init(
             upstream: url,
             allowed_origins: allowed_origins,
             proxy_url: "#{cors_scheme}://#{cors_host}:#{cors_port}",
-            # Since we replaced the host with the IP address in ip_url above, we need to force the host
+            # We need to force the host
             # used for ssl verification here so that the connection isn't rejected.
             # Note that we have to convert the authority to a charlist, since this uses Erlang's `ssl` module
             # internally, which expects a charlist.
             client_options: [
-              ssl: [{:server_name_indication, to_charlist(authority)}, {:versions, [:"tlsv1.2",:"tlsv1.3"]}]
-            ],
-            # preserve_host_header: true
+              ssl: [
+                {:server_name_indication, to_charlist(authority)},
+                {:versions, [:"tlsv1.2", :"tlsv1.3"]}
+              ]
+            ]
           )
 
         body = ReverseProxyPlug.read_body(conn)
         is_head = conn |> Conn.get_req_header("x-original-method") == ["HEAD"]
 
-        %Conn{}|> Map.merge(conn)
-        |> Map.put(
-          :method,
-          if is_head do
-            "HEAD"
-          else
-            conn.method
+        try do
+          # First, make a HEAD request to check for redirects using HTTPoison
+          case HTTPoison.head(url, [],
+                 follow_redirect: false,
+                 ssl: [
+                   {:server_name_indication, to_charlist(authority)},
+                   {:versions, [:"tlsv1.2", :"tlsv1.3"]}
+                 ],
+                 timeout: 15_000,
+                 recv_timeout: 15_000
+               ) do
+            {:ok, %HTTPoison.Response{status_code: status_code, headers: headers}}
+            when status_code in [301, 302, 303, 307, 308] ->
+              # Found a redirect
+              location_header =
+                headers
+                |> Enum.find(fn {k, _v} -> String.downcase(k) == "location" end)
+                |> elem(1)
+
+              if location_header do
+                # Resolve relative URLs against the current URL
+                redirect_url = URI.merge(uri, location_header) |> URI.to_string()
+                cors_proxy_with_redirects(conn, redirect_url, redirect_count + 1)
+              else
+                Logger.warning("CORS Proxy: Redirect response missing location header")
+                # Fall back to ReverseProxyPlug for this response
+                make_reverse_proxy_request(conn, url, body, is_head, opts)
+              end
+
+            {:ok, %HTTPoison.Response{status_code: status_code}} ->
```

Comment:
This is a success, so we don't care what the exact status code is, so this line should be:
```
{:ok, %HTTPoison.Response{}} ->
```